website/utils/url_utils.ts

```diff
@@ -2,16 +2,47 @@ import FirebaseRequest, { DatasetMetaData } from "../../public/firebase";
 
 import { AppProps, GlobalViewerSettings } from "../../src/aics-image-viewer/components/App/types";
 import { ViewMode } from "../../src/aics-image-viewer/shared/enums";
-import { ViewerChannelSettings } from "../../src/aics-image-viewer/shared/utils/viewerChannelSettings";
+import {
+  ViewerChannelSetting,
+  ViewerChannelSettings,
+} from "../../src/aics-image-viewer/shared/utils/viewerChannelSettings";
+
+const CHANNEL_STATE_KEY_REGEX = /^c[0-9]+$/;
+// Match colon-separated pairs of alphanumeric strings
+const LUT_REGEX = /^[a-z0-9.]*:[ ]*[a-z0-9.]*$/;
+const HEX_COLOR_REGEX = /^[0-9a-fA-F]{6}$/;
+
+export type ViewerChannelSettingJson = {
+  /** Color, as a 6-digit hex color */
+  c?: string;
+  /** Colorize. "1" is enabled. */
+  cz?: "1" | "0";
+  /** Colorize alpha, in the [0, 1] range. */
+  cza?: string;
+  /** Opacity, in the [0, 1 range] */
+  op?: string;
+  /** LUT to map from intensity to opacity. Should be two alphanumeric values separated
+   * by a colon, e.g. "0:255", "p50:p90", or "autoij:0".
+   */
+  lut?: string;
+  /** Volume enabled. "1" is enabled. */
+  v?: "1" | "0";
+  /** Isosurface enabled. "1" is enabled. */
+  i?: "1" | "0";
+  /** Isosurface value, in the [0, 255] range. */
+  iv?: string;
+};
 
 const paramKeys = ["mask", "ch", "luts", "colors", "url", "file", "dataset", "id", "view"];
+type ChannelKey = `c${number}`;
+
 type ParamKeys = (typeof paramKeys)[number];
-type Params = { [_ in ParamKeys]?: string };
+type Params = { [_ in ParamKeys | ChannelKey]?: string };
 
-function urlSearchParamsToParams(searchParams: URLSearchParams): Params {
+export function urlSearchParamsToParams(searchParams: URLSearchParams): Params {
   const result: Params = {};
   for (const [key, value] of searchParams.entries()) {
-    if (paramKeys.includes(key)) {
+    if (paramKeys.includes(key) || CHANNEL_STATE_KEY_REGEX.test(key)) {
```

Comment:
Nit: consider separating `ch, luts, colors` out of paramKeys so we can check against mixing the two types of per-channel settings?    Would that help keep things more organized?

Reply:
Let me think on this one! I'll at least move them into their own list (like `deprecatedParamKeys`).